### PR TITLE
Prepare for SQLAlchemy 2

### PR DESCRIPTION
**Pull request by msoucy**

### CODE REVIEW

File: signinapp/event.py [hunk 1]
```diff
@@ -23,7 +24,9 @@ def scan():
     event = request.values['event']
     name = request.values['name']
 
-    ev: Event = Event.query.filter_by(code=event, enabled=True).one_or_none()
+    ev: Event = db.session.execute(
```
Comment (bot190) [inline signinapp/event.py]: Any reason not to use `scalar` instead of `execute` here, and elsewhere?

File: signinapp/model.py [hunk 2]
```diff
@@ -48,6 +47,13 @@ def canonical_name(name: str):
         return mk_hash(f"{m['first']} {m['last']}")
 
 
+def get_form_ids(model, prefix=False):
```
Comment (bot190) [inline signinapp/model.py]: `prefix` does not do at all what I was expecting.. Maybe `add_none_option` or something?

File: signinapp/model.py [hunk 3]
```diff
@@ -67,6 +73,38 @@ def get_size_names(cls):
...
+    description = db.Column(db.String)
+    icon = db.Column(db.String)
+    color = db.Column(db.String, default="black")
+
+    awards = db.relationship("BadgeAward", back_populates="badge")
+
+    @classmethod
+    def from_name(cls, name) -> 'Subteam':
+        ' Get a badge by name '
+        return cls.query.filter_by(name=name).one_or_none()
```
Comment (bot190) [inline signinapp/model.py]:
```suggestion
        return db.session.scalar(select(cls).filter_by(name=name))
```

File: signinapp/templates/profile.html.jinja2 [hunk 4]
```diff
@@ -54,10 +54,10 @@
             <h2>Badges:</h2>
             <ul>
               {% if user.badges %}
-                {% for award in user.badges|sort(attribute="name") %}
+                {% for b in user.badges|sort(attribute="name") %}
                   <li>
                     <a class="link-light"
-                       href="{{ url_for('user.badge', badge_id=award.badge_id)}}">{{ show_badge(award.badge)}}</a>
+                       href="{{ url_for('user.badge', badge_id=b)}}">{{ show_badge(b)}}</a>
```
Comment (bot190) [inline signinapp/templates/profile.html.jinja2]: Does this give the correct URL still?

**Review by bot190 — CHANGES_REQUESTED** [review]

File: signinapp/templates/profile.html.jinja2 (hunk 4 above)
Comment (msoucy) [reply signinapp/templates/profile.html.jinja2]: ..... no.

Comment (msoucy) [reply signinapp/templates/profile.html.jinja2]: Yes

**Review by bot190 — APPROVED** [review]

**Pull request merged**